arangod/RocksDBEngine/RocksDBCollectionMeta.cpp

```diff
@@ -202,6 +202,7 @@ Result RocksDBCollectionMeta::serializeMeta(rocksdb::WriteBatch& batch,
   } else {  // maxCommitSeq is == UINT64_MAX without any blockers
     appliedSeq = std::min(appliedSeq, maxCommitSeq);
   }
+  TRI_ASSERT(commitSeq <= appliedSeq);
```

Comment:
I think what you actually want is `TRI_ASSERT(appliedSeq <= commitSeq)` inside the `if` branch of the statement before the assignment.

Reply:
hm shouldn't it hold regardless ?